src/ssl/MemStats.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
```

Comment:
I checked that GnuTLS [has](https://man7.org/linux/man-pages/man3/gnutls_global_set_mem_functions.3.html) an analog of CRYPTO_set_mem_functions() but it is deprecated now. So it looks like this code is openssl-specific and belongs here (not the generic 'security' folder).